lib/Sema/ImportResolution.cpp

```diff
@@ -264,24 +276,19 @@ class ImportResolver final : public DeclVisitor<ImportResolver> {
 ///
 /// Import resolution operates on a parsed but otherwise unvalidated AST.
 void swift::performImportResolution(SourceFile &SF) {
-  FrontendStatsTracer tracer(SF.getASTContext().Stats,
-                             "Import resolution");
-
-  // Make sure we skip adding the standard library imports if the
-  // source file is empty.
-  if (SF.ASTStage == SourceFile::ImportsResolved ||
-      SF.getTopLevelDecls().empty()) {
-    SF.ASTStage = SourceFile::ImportsResolved;
+  // If we've already performed import resolution, bail.
+  if (SF.ASTStage == SourceFile::ImportsResolved)
```

Comment:
I've always wondered why these stage guards aren't assertions.

Reply:
Good idea! Fortunately this can all go away once import resolution is requestified.

Reply:
Ah nevermind, the frontend can currently call `performImportResolution` before calling `performTypeChecking` (which also calls `performImportResolution`). I'll leave this as-is, and rip it out when requestifying import resolution.